gcloud/_helpers.py

```diff
@@ -430,20 +479,20 @@ def _datetime_to_pb_timestamp(when):
 def _name_from_project_path(path, project, template):
     """Validate a URI path and get the leaf object's name.
 
-    :type path: string
+    :type path: str
     :param path: URI path containing the name.
 
-    :type project: string or NoneType
+    :type project: str or NoneType
     :param project: The project associated with the request. It is
                     included for validation purposes.  If passed as None,
                     disables validation.
 
-    :type template: string
+    :type template: str
     :param template: Template regex describing the expected form of the path.
                      The regex must have two named groups, 'project' and
                      'name'.
 
-    :rtype: string
+    :rtype: str
```

Comment:
All of these type changes are fine, but they add a lot of noise to this PR and have nothing to do with the PR. In the future please separate concerns.